accepted/future-releases/abstract-external-fields/feature-specification.md

```diff
@@ -0,0 +1,205 @@
+# Abstract Fields and External Fields
+
+Authors: lrn@google.com, eernst@google.com<br>
+Version: 1.0
+
+## Background and Motivation
+
+Dart allows abstract instance methods, getters and setters to be declared
+in classes and mixins.  It allows external functions, methods, getters, and
+setters to be declared as top-level, static or instance declarations, and
+it allows external constructors to be declared in classes.
+
+The syntax of an abstract member is simply a declaration with no body.  The
+syntax for an external declaration prefixes the declaration with the
+modifier `external`.
+
+Abstract declarations add members to a class interface, allowing statically
+checked usage of different implementations provided in subclasses.
+External declarations add members to a namespace (such as a class interface
+or a library namespace), also allowing statically checked usage of
+implementations that are not known statically; the declarations are
+associated with implementations via an implementation specific mechanism.
+
+A field is associated with a storage location and an implicitly induced
+setter and/or getter which amounts to an implementation, so fields seem to
+be inherently non-abstract and non-external. However, we can identify the
+field with its accessors:
+
+1. An "abstract instance field" can be a pair of an abstract instance
```

Comment:
This is the background and motivation section, and this particular part is about the design and the underlying ideas, so it doesn't actually define anything. I've adjusted the wording to make this more obvious.

Reply:
I am firm believer that the background and motivation should not talk about the *new* thing at all.

First describe the status quo, and why it has problems (Background and Motivation). Preferably briefly, since it can easily be known to the reader already.

Then describe the proposed change. Be precise and specific. This is specification.

Then elaborate on consequences, alternatives and justifications for choices.
If you need to justify the design, do it here, not where it's being specified. That might cause readers to start wondering about alternatives that won't happen anyway.

Reply:
> should not talk about the _new_ thing at all.

That's indeed a nice principle. This particular section was written by you, and I updated it, but the mixture of old and new was there all the time. Anyway, I like the structure you suggest, and I've revised the text to separate the two.